bigbluebutton-html5/imports/ui/components/layout/layout-manager/presentationOnlyLayout.jsx

```diff
@@ -0,0 +1,381 @@
+
+  const prevDeviceType = usePrevious(deviceType);
+
+  const calculatesMediaBounds = (mediaAreaBounds, sidebarSize) => {
+    const mediaBounds = {};
+    const { element: fullscreenElement } = fullscreen;
+
+    if (
+      fullscreenElement === 'Presentation'
+      || fullscreenElement === 'Screenshare'
+      || fullscreenElement === 'ExternalVideo'
+    ) {
+      mediaBounds.width = windowWidth();
+      mediaBounds.height = windowHeight();
+      mediaBounds.top = 0;
+      mediaBounds.left = !isRTL ? 0 : null;
+      mediaBounds.right = isRTL ? 0 : null;
+      mediaBounds.zIndex = 99;
+      return mediaBounds;
+    }
+
+    mediaBounds.height = mediaAreaBounds.height;
+    mediaBounds.width = mediaAreaBounds.width;
+    mediaBounds.top = DEFAULT_VALUES.navBarHeight + bannerAreaHeight();
+    mediaBounds.left = !isRTL ? mediaAreaBounds.left : null;
+    mediaBounds.right = isRTL ? sidebarSize : null;
+    mediaBounds.zIndex = 1;
+
+    return mediaBounds;
+  };
+
+  const calculatesSidebarContentHeight = () => ({
+    height: 0,
+    minHeight: 0,
+    maxHeight: 0,
+  });
+
+  const calculatesCameraDockBounds = () => {
+    const cameraDockBounds = {};
+
+    cameraDockBounds.top = 0;
+    cameraDockBounds.left = 0;
+    cameraDockBounds.right = 0;
+    cameraDockBounds.minWidth = 0;
+    cameraDockBounds.width = 0;
+    cameraDockBounds.maxWidth = 0;
+    cameraDockBounds.minHeight = 0;
+    cameraDockBounds.height = 0;
+    cameraDockBounds.maxHeight = 0;
+
+    return cameraDockBounds;
+  };
+
+  const calculatesLayout = () => {
+    const {
+      calculatesNavbarBounds,
+      calculatesActionbarBounds,
+      calculatesSidebarNavWidth,
+      calculatesSidebarNavHeight,
+      calculatesSidebarNavBounds,
+      calculatesSidebarContentWidth,
+      calculatesSidebarContentBounds,
+      calculatesMediaAreaBounds,
+      isTablet,
+    } = props;
+    const { captionsMargin } = DEFAULT_VALUES;
+
+    const sidebarNavWidth = calculatesSidebarNavWidth();
+    const sidebarNavHeight = calculatesSidebarNavHeight();
+    const sidebarContentWidth = calculatesSidebarContentWidth();
+    const sidebarNavBounds = calculatesSidebarNavBounds();
+    const sidebarContentBounds = calculatesSidebarContentBounds(sidebarNavWidth.width);
+    const mediaAreaBounds = calculatesMediaAreaBounds(
+      sidebarNavWidth.width,
+      sidebarContentWidth.width,
+    );
+    const navbarBounds = calculatesNavbarBounds(mediaAreaBounds);
+    const actionbarBounds = calculatesActionbarBounds(mediaAreaBounds);
+    const sidebarSize = sidebarContentWidth.width + sidebarNavWidth.width;
+    const mediaBounds = calculatesMediaBounds(mediaAreaBounds, sidebarSize);
+    const sidebarContentHeight = calculatesSidebarContentHeight();
+    const cameraDockBounds = calculatesCameraDockBounds(
+      mediaBounds,
+      mediaAreaBounds,
+      sidebarNavWidth.width,
+      sidebarContentWidth.width,
+      sidebarContentHeight.height,
+    );
```

Comment:
This function expects no arguments

```suggestion
    const cameraDockBounds = calculatesCameraDockBounds();
```